articles/native-platforms/ios-objc/03-session-handling.md

```diff
@@ -0,0 +1,131 @@
+
+#### ii. Add the SimpleKeychain dependency
+
+
+We're going to use the [SimpleKeychain](https://github.com/auth0/SimpleKeychain) library to help us manage user credentials. Make sure you integrate it before proceeding.
+
+##### a. Carthage
+
+If you are using Carthage, add the following line to the `Cartfile`:
+
+```ruby
+github "auth0/SimpleKeychain"
+```
+
+Then, run `carthage bootstrap`.
+
+> For more information about Carthage usage, check [their official documentation](https://github.com/Carthage/Carthage#if-youre-building-for-ios-tvos-or-watchos).
+
+##### b. Cocoapods
+
+If you are using [Cocoapods](https://cocoapods.org/), add these lines to your `Podfile`:
+
+```ruby
+pod 'SimpleKeychain', '~> 0.7'
+```
+
+Then, run `pod install`.
+
+> For further reference on Cocoapods, check [their official documentation](http://guides.cocoapods.org/using/getting-started.html).
+
+
+### 1. Save the token for later
+
+First of all, you need to import the headers for Lock and SimpleKeychain
+
+```objc
+#import <Lock/Lock.h>
+#import "SimpleKeychain.h"
+```
+
+Once your user successfully signs in, you'll have a reference to a 'A0Token' object with the user's credentials. SimpleKeychain is a simple 'key-value' storage that we can use keep safe the the user's 'id_token'.
+
+> The `idToken` is a string representing, basically, the user's [JWT token](https://en.wikipedia.org/wiki/JSON_Web_Token).
+
+```objc
+- (void) saveCredentials:(A0Token* ) token
+{
+    A0SimpleKeychain* keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
+    [keychain setString:token.idToken forKey:@"id_token"];
+    [keychain setString:token.refreshToken forKey:@"refresh_token"];
+}
+```
+
+We'll be storing the 'refresh_token' too, which we'll use to get a new token, once the current one has expired.
+
+
+### 2. Don't I know you from somewhere?
+
+Once you have stored the user's token, next time the app launches, you can use it to restore the user's profile and avoid making the user have to sign in again. First, you need to check if SimpleKeychain has a token stored:
+
+```objc
+    A0SimpleKeychain* keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
+    
+    if([keychain stringForKey:@"id_token"]){
+    	// There is a token stored
+
+    	[[A0Lock sharedLock].apiClient fetchUserProfileWithIdToken:[keychain stringForKey:@"id_token"] 
+    	success:^(A0UserProfile * _Nonnull profile) {
+        	// You have successfully retreived the user's profile, you don't need to sign in again.
+        	// Let your user continue to the next step
+		} failure:^(NSError * _Nonnull error) {
+			// Something went wrong, let the user know
+		}
+    }
+```
+
+If the `fetchUserProfileWithIdToken:` call is successful, you can continue as if the user has singned in. But if it failed there could be a number of reasons for it. One option is that it has expired, in which case you still have the option to refresh it, and keep the session valid:
```

Comment:
- singned → signed
- has signed in → had signed in
- But if it failed there could be a number of reasons for it. → But, if it failed, there could be a number of reasons for it.